### PR TITLE
Send to endpoints in sorted order.  Sending to AMQP prioritized first…

### PR DESCRIPTION
**Pull request by aberke**
…, before SIP.

### CODE REVIEW

**Comment by aberke** [comment]
As discussed with @lazedo.
Note I can't truly test this locally...

**Comment by fenollp** [comment]
Hey @aberke, thanks for the nice work.
Please make sure you are not introducing whitespace in the codebase: `applications/doodle/src/module/cf_sms_user.erl`

**Comment by lazedo** [comment]
@aberke here's what i usually do `sed -i 's/\s*$//g' src/*.erl`

File: applications/doodle/src/module/cf_sms_user.erl [hunk 1]
```diff
@@ -91,5 +90,21 @@ get_endpoints(UserId, Data, Call) ->
                         end
                 end
                 ,{[], 0}
-                ,cf_attributes:owned_by(UserId, <<"device">>, Call)
-               ).
+                ,EndpointIds
+               ),
+    SortedEndpoints = sort_endpoints_by_type(Endpoints),
+    {SortedEndpoints, DndCount}.
+
```
Comment (aberke) [inline applications/doodle/src/module/cf_sms_user.erl]: @fenollp is this the whitespace that you're referring to?  I'll amend my commit in the style you prefer.

**Comment by aberke** [comment]
@fenollp I amended the commit with whitespace removed.

File: applications/doodle/src/module/cf_sms_user.erl [hunk 2]
```diff
@@ -91,5 +90,20 @@ get_endpoints(UserId, Data, Call) ->
...
+sort_endpoints_by_type(Endpoints) ->
+    lists:sort(fun(EndpointA, EndpointB) ->
+            EndpointAValue = endpoint_type_sort_value(wh_json:get_value(<<"Endpoint-Type">>, EndpointA)),
+            EndpointBValue = endpoint_type_sort_value(wh_json:get_value(<<"Endpoint-Type">>, EndpointB)),
+            (EndpointAValue < EndpointBValue)
+        end,
+        Endpoints).
+
```
Comment (fenollp) [inline applications/doodle/src/module/cf_sms_user.erl]: @aberke there's some left here :)

**Comment by lazedo** [comment]
@aberke change to doodle folder and do `sed -i 's/\s*$//g' src/*.erl`

**Comment by aberke** [comment]
@fenollp @lazedo I believe I removed all the extra whitespace.  I also added `-spec`'s

File: applications/doodle/src/module/cf_sms_user.erl [hunk 3]
```diff
@@ -91,5 +92,22 @@ get_endpoints(UserId, Data, Call) ->
...
                 end
                 ,{[], 0}
-                ,cf_attributes:owned_by(UserId, <<"device">>, Call)
-               ).
+                ,EndpointIds
+               ),
+    SortedEndpoints = sort_endpoints_by_type(Endpoints),
+    {SortedEndpoints, DndCount}.
+
+-spec sort_endpoints_by_type([wh_json:object()]) -> [wh_json:object()].
```
Comment (lazedo) [inline applications/doodle/src/module/cf_sms_user.erl]: use wh_json:objects() instead of [wh_json:object()]

**Comment by aberke** [comment]
using wh_json:objects() now @lazedo

**Comment by lazedo** [comment]
@aberke Thanks. cherry-picked to 3.21/3.22

**Pull request merged**